### PR TITLE
Census: more data updates

### PR DESCRIPTION
**Pull request by breville**
We've received corrections and updates to the data added last week.  This PR incorporates the most important 5 states' updates.

- CA - delete 2018 entries, ingest first 2017 entries
- HI - delete 2018 entries, ingest first 2017 entries
- IN - re-ingest, it's the same data with some new schools added
- MT - re-ingest, it's the same data with some new schools added
- RI - delete 2018 entries, ingest new 2017 entries (while leaving existing 2017 entries intact)

For the cases where we need to delete 2018 entries, we're going to first manually delete them using the following:

```
Census::StateCsOffering.where(school_year: 2018).where('state_school_id LIKE ?', 'CA%').delete_all
Census::StateCsOffering.where(school_year: 2018).where('state_school_id LIKE ?', 'HI%').delete_all
Census::StateCsOffering.where(school_year: 2018).where('state_school_id LIKE ?', 'RI%').delete_all
```

Deployment plan: after this code is live on production, we'll run the above Ruby code to delete entries, and will delete the associated files from S3, then we'll copy the new CSV files from this PR to S3, and then manually seed this data on staging and production using the dashboard console.

In hindsight, it would be nice to get all of this data under source control, and to re-seed it from scratch with every deploy, so that we are closer to having an immutable data collection with a single direction of data flow.

### CODE REVIEW

**Comment by sureshc** [comment]
This data maintenance indicates to me that the seeding logic should delete all entries for the current file's state and year before importing the file.  That's the pattern we use for the Pegasus database content that is imported from Google Sheets.  This approach would allow us to replace a CSV with a correction version in S3 instead of using version numbers at the end of file names.

File: dashboard/app/models/census/state_cs_offering.rb [hunk 1]
```diff
@@ -76,13 +76,17 @@ class Census::StateCsOffering < ApplicationRecord
...
-  # states in 2018-19, as listed here.  The expectation is that all states will use
-  # the new format as of 2019-20.
+  # The following states use the "V2" format for CSV files in 2017-2018.
+  # (The expectation is that all states will use the new format as of 2019-20.)
+  STATES_USING_FORMAT_V2_IN_2017_18 = %w(
+    CA
+    HI
+  ).freeze
+
+  # The following states use the "V2" format for CSV files in 2017-2018.
```
Comment (sureshc) [inline dashboard/app/models/census/state_cs_offering.rb]:
```suggestion
  # The following states use the "V2" format for CSV files in 2018-2019.
```

**Comment by breville** [comment]
@sureshc I believe that is a good long-term plan.  However there are two areas of fragility here: 
- first is that we are assuming we can always recreate this data from our CSVs in S3, and while this is probably true, it feels fragile, especially since we didn't even have revision tracking enabled on this bucket until a week ago.  I would hate to lose any of what is turning out to be hand-crafted data.
- second is that we don't actually store the state by itself in the model, and would have to infer it from the state school ID or the NCES ID.  We are fortunate in deleting the newly-added 2018 data because we enforced the requirement of a state school ID in the data, and they always begin with the state code, but this gets more fragile and expensive if we have to infer the state from the NCES ID for every delete operation.

File: dashboard/app/models/census/state_cs_offering.rb [hunk 2]
```diff
@@ -117,24 +121,39 @@ class Census::StateCsOffering < ApplicationRecord
...
   UPDATES_FOR_STATES_USING_FORMAT_V2_IN_MID_2018_19 = {
     DC: 2,
     DE: 2,
-    HI: 2,
     NH: 2,
     NJ: 2,
     VT: 2
   }.freeze
 
   def self.state_uses_format_v2(state_code, school_year, update)
```
Comment (sureshc) [inline dashboard/app/models/census/state_cs_offering.rb:144]: This method is starting to get complicated.  I can follow the logic because I helped design this approach.  I'm now wondering whether putting the format in the filename (`LA/2018-2019.v2.csv`) and not having multiple files per year would be easier for other engineers to maintain in the future.

Comment (islemaster) [reply dashboard/app/models/census/state_cs_offering.rb:144]: I like that idea!  Making a stronger assertion about format as part of the filename makes a lot of sense.

**Review by islemaster — APPROVED** [review]
Plan makes sense to me.

File: dashboard/app/models/census/state_cs_offering.rb (hunk 2 above)
Comment (breville) [reply dashboard/app/models/census/state_cs_offering.rb:144]: It's a good point, and something I considered last week.  The main reason I decided not to was because we would have two forms of versioning in the one filename - one for the format, and one to declare that this was an incremental update for the same year - and the latter was the long-term investment while the former was a transitional measure.  Best would be to store the format version as metadata in the file, but unfortunately CSV doesn't lend itself to having additional metadata.  I've also pondered using a heuristic to determine the format version, such as the presence of certain named fields, but that also feels fragile if we want to update the format version for other reasons.

Long story short, I agree that the complexity is on the tipping point of needing a refactor.  That said, I'm hopeful that this is the end of of ingestions for the current year, and that everything gets simpler for 2019-20 and beyond when everything is in the new format.

**Comment by breville** [comment]
This is a followup to last week's work in https://github.com/code-dot-org/code-dot-org/pull/29710 and its linked PRs.
